assignment_4/main.erl

```diff
@@ -69,25 +72,25 @@ server3() ->
     server3(0).
 
 start() -> 
-    Pid1 = spawn(main, server1, []),
-    Pid2 = spawn(main, server2, []),
-    Pid3 = spawn(main, server3, []),
+    Pid1 = spawn(fun server1/0),
+    Pid2 = spawn(fun server2/0),
+    Pid3 = spawn(fun server3/0),
     register(server1, Pid1),
     register(server2, Pid2),
     register(server3, Pid3),
-    Pid1 ! {add, {1, 2}},
-    Pid1 ! {sub, {3, 4}},
-    Pid1 ! [1,2,3,4],
-    Pid1 ! {error, "Error!"},
-    Pid1 ! {unhandled, "unhandled"},
     main().
 
 get_numData() ->
     {ok, Work} = io:read("Enter a command: "),
-    Work.
+    case Work of
+        halt -> halt;  % Return 'halt'
```

Comment:
why do we need to do this?

Reply:
I was trying to get it to stop printing halt by itself when I enter halt as an input. Not sure if it actually works tho